spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/abstractnn/AbstractModule.scala

```diff
@@ -580,5 +602,23 @@ abstract class AbstractModule[A <: Activity: ClassTag, B <: Activity: ClassTag,
               ): Array[(ValidationResult, ValidationMethod[T])] = {
     Validator(this, dataSet).test(vMethods)
   }
+
+  private var nameToSubModules: Map[String, Module[T]] = _
+  /**
+   * get all submodules map, use the module name as key
+   */
+  def getSubModules(): Map[String, Module[T]] = {
+    if (nameToSubModules == null) nameToSubModules = Utils.getNamedModules(this)
+    nameToSubModules
+  }
+
+  /**
+   * get submodule by module name
+   */
+  def getSubModule(name: String): Module[T] = {
+    getSubModules()
+    require(nameToSubModules.contains(name), s"$name does not exists")
+    nameToSubModules(name)
```

Comment:
Why don't you use the apply method but write a new one?

model("layer1") will give you the corrrectLayer